Fix hysteresis support in gov_step_wise.c

jkreileder opened the pull request:
Directly get hyst value instead of going through an optional and, now, unimplemented function.

Fixes https://github.com/raspberrypi/linux/issues/5726
(Note that there are more drivers with this problem. But this PR fixes it for stand Pi fans.)

jkreileder committed:
Fix hysteresis support in gov_step_wise.c

Directly get hyst value instead of going through an
optional and, now, unimplemented function.

Signed-off-by: Jürgen Kreileder <jk@blackdown.de>

pelwell commented on drivers/thermal/gov_step_wise.c:
```diff
@@ -99,12 +99,15 @@ static void thermal_zone_trip_update(struct thermal_zone_device *tz, int trip_id
-	}
 	trip_type = trip.type;
 
+	if (trip.hysteresis)
```
I don't see why you've put this code here - the location of the old version is more logical.

pelwell commented on drivers/thermal/gov_step_wise.c:
```diff
@@ -99,12 +99,15 @@ static void thermal_zone_trip_update(struct thermal_zone_device *tz, int trip_id
 
+	if (trip.hysteresis)
+		hyst_temp = trip_temp - trip.hysteresis;
+	else
```
I don't think we need to warn about a zero hysteresis - what's wrong with this?:
```
    trip_temp = trip.temperature;
    hyst_temp = trip_temp - trip.hysteresis;
```

jkreileder replied:
Nothing really, the only reason for this was to make it more like the existing code in gov_bang_bang.c: https://github.com/torvalds/linux/blob/9b6de136b5f0158c60844f85286a593cb70fb364/drivers/thermal/gov_bang_bang.c#L22-L24

I'll make a simplified PR later.

jkreileder replied:
Updated now.

pelwell reviewed (commented):
Thanks for the patch - the new mechanism is much simpler.

jkreileder committed:
Fix hysteresis support in gov_step_wise.c

Directly get hyst value instead of going through an
optional and, now, unimplemented function.

Signed-off-by: Jürgen Kreileder <jk@blackdown.de>

pelwell commented:
Thanks!
